DiagnosticsExtension/Models/ConnectionStringValidator/QueueStorageValidator.cs

```diff
@@ -0,0 +1,199 @@
+            string appSettingClientIdValue, appSettingClientCredValue = null;
+            string value = null;
+            QueueServiceClient client = null;
+
+            if (envDict.Contains(appSettingName))
+            {
+                try
+                {
+                    value = Environment.GetEnvironmentVariable(appSettingName);
+                    client = new QueueServiceClient(value);
+                }
+                catch (ArgumentNullException e)
+                {
+                    throw new EmptyConnectionStringException(e.Message, e);
+                }
+                catch (Exception e)
+                {
+                    throw new MalformedConnectionStringException(e.Message, e);
+                }
+
+            }
+            else
+            {
+                try
+                {
+
+
+                    value = Environment.GetEnvironmentVariable(appSettingName + "__queueServiceUri");
+                    if (!string.IsNullOrEmpty(value))
+                    {
+                        appSettingClientIdValue = Environment.GetEnvironmentVariable(appSettingName + "__clientId");
+                        appSettingClientCredValue = Environment.GetEnvironmentVariable(appSettingName + "__credential");
+                        Uri objuri = new Uri(value);
+                        if (!string.IsNullOrEmpty(appSettingClientIdValue))
+                        {
+                            if (appSettingClientCredValue != "managedidentity")
+                            {
+                                throw new ManagedIdentityException("managedidentitymissed");
+                            }
+                            else
+                            {
+                                identityType = ConnectionStringValidationResult.ManagedIdentityType.User;
+                                client = new QueueServiceClient(objuri, new Azure.Identity.ManagedIdentityCredential(appSettingClientIdValue));
+                            }
+
+                        }
+                        else
+                        {
+                            identityType = ConnectionStringValidationResult.ManagedIdentityType.System;
+                            client = new QueueServiceClient(objuri, new Azure.Identity.ManagedIdentityCredential());
+                        }
+                    }
+                    else
+                    {
+                        throw new ManagedIdentityException("fullyQualifiedNamespacemissed");
+                    }
+                }
+                catch (Exception e)
+                {
+                    throw new ManagedIdentityException(e.Message, e);
+                }
+            }
+            client.GetQueuesAsync();
+            var resultSegment =
+            client.GetQueues(QueueTraits.Metadata, null, default)
+            .AsPages(default, 10);
+            foreach (Azure.Page<QueueItem> containerPage in resultSegment)
```

Comment:
It does not look like we are doing any validation here so why iterate over the pages and queues?

Reply:
We need to call the GetQueues and then iterate through the page for the authentication to come into picture. Just by creating the client, the authentication does not come into picture.